docs/report/main.tex

```diff
@@ -825,12 +851,125 @@ \subsection{How The Algorithm Works}
+\draw    (595,149.5) -- (625,149.5)(595,152.5) -- (625,152.5) ;
+%Straight Lines [id:da9226246067589841] 
+\draw    (595,189.5) -- (625,189.5)(595,192.5) -- (625,192.5) ;
+%Shape: Brace [id:dp5786954336630095] 
+\draw   (208.13,205) .. controls (208.15,209.67) and (210.49,211.99) .. (215.16,211.98) -- (357.81,211.53) .. controls (364.48,211.51) and (367.82,213.83) .. (367.83,218.5) .. controls (367.82,213.83) and (371.14,211.49) .. (377.81,211.47)(374.81,211.48) -- (528.16,210.99) .. controls (532.83,210.98) and (535.15,208.64) .. (535.13,203.97) ;
+%Shape: Brace [id:dp3814057437492582] 
+\draw   (470.13,127) .. controls (470.13,122.33) and (467.8,120) .. (463.13,120) -- (410.13,120) .. controls (403.46,120) and (400.13,117.67) .. (400.13,113) .. controls (400.13,117.67) and (396.8,120) .. (390.13,120)(393.13,120) -- (334.13,120) .. controls (329.46,120) and (327.13,122.33) .. (327.13,127) ;
+%Shape: Brace [id:dp5248227356988384] 
+\draw   (66.13,152) .. controls (61.46,152) and (59.13,154.33) .. (59.13,159) -- (59.13,162.16) .. controls (59.13,168.83) and (56.8,172.16) .. (52.13,172.16) .. controls (56.8,172.16) and (59.13,175.49) .. (59.13,182.16)(59.13,179.16) -- (59.13,185) .. controls (59.13,189.67) and (61.46,192) .. (66.13,192) ;
+%Straight Lines [id:da6246903517915476] 
+\draw  [dash pattern={on 0.84pt off 2.51pt}]  (75.13,168) -- (133.13,168) ;
+%Straight Lines [id:da7651260164716407] 
+\draw  [dash pattern={on 0.84pt off 2.51pt}]  (251,168) -- (305.13,168) ;
+%Straight Lines [id:da5357419850229155] 
+\draw  [dash pattern={on 0.84pt off 2.51pt}]  (504,167) -- (592.13,167) ;
+
+% Text Node
+\draw (151.6,139.4) node [anchor=north west][inner sep=0.75pt]    {$H$};
+% Text Node
+\draw (151.6,181.4) node [anchor=north west][inner sep=0.75pt]    {$H$};
+% Text Node
+\draw (209,155.4) node [anchor=north west][inner sep=0.75pt]    {$U_{\omega }$};
+% Text Node
+\draw (273.6,139.4) node [anchor=north west][inner sep=0.75pt]    {$H$};
+% Text Node
+\draw (273.6,181.4) node [anchor=north west][inner sep=0.75pt]    {$H$};
+% Text Node
+\draw (513.6,137.4) node [anchor=north west][inner sep=0.75pt]    {$H$};
+% Text Node
+\draw (513.6,179.4) node [anchor=north west][inner sep=0.75pt]    {$H$};
+% Text Node
+\draw (346,153.4) node [anchor=north west][inner sep=0.75pt]    {$2\ket{0^{n}}\bra{0^{n}} \ -\ I^{n}$};
+% Text Node
+\draw (68,142.4) node [anchor=north west][inner sep=0.75pt]    {$\ket{0}$};
+% Text Node
+\draw (68,181.4) node [anchor=north west][inner sep=0.75pt]    {$\ket{0}$};
+% Text Node
+\draw (16,162.4) node [anchor=north west][inner sep=0.75pt]    {$\ket{0^{n}}$};
+% Text Node
+\draw (316,95) node [anchor=north west][inner sep=0.75pt]   [align=left] {Grover's Diffusion Operator};
+% Text Node
+\draw (313,213) node [anchor=north west][inner sep=0.75pt]   [align=left] {Repeat $\displaystyle \approx \ \frac{\pi }{4}\sqrt{N}$ times};
+
+
+\end{tikzpicture}
 \vspace{5mm}
 
+\textbf{Quantum circuit representation of Grover's Algorithm.}
```

Comment:
For a follow up, but we should figure out how to get LaTeX to auto number our figures.

Reply:
Ok I will look into this .